x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -190,6 +190,6 @@ public static TypeResolution typeMustBe(Expression e,
                     paramOrd == null || paramOrd == ParamOrdinal.DEFAULT ? "" : " " + paramOrd.name().toLowerCase(Locale.ROOT),
                     Strings.arrayToDelimitedString(acceptedTypes, " or "),
                     Expressions.name(e),
-                    e.dataType().esType));
+                    e.dataType().esType.toUpperCase(Locale.ROOT)));
```

Comment:
I think we should keep the lower casing in place since that's how the types are defined in ES as well (`keyword` vs `KEYWORD`).
If anything, we should change for example the column descriptors to always return the ES types in lowercase to make the context (SQL type vs ES type) even clearer.